### PR TITLE
fix: Invalid Email address gets taken in in Attendee Details.

### LINKED ISSUE #865: Invalid Email address gets taken in in Attendee Details.
**Describe the bug**

Invalid Email address gets accepted in Email Edittext of the Attendee details while filling out the details of ticket holders

**To Reproduce**

1. Checkout to the Attendee Fragment.
2. Enter email in the email section.

**Expected behavior**

Only valid email address should be accepted.

**Logs**

**Screenshots**

email_invalid

**Smartphone Info:**

|               | |
|---------------|-|
|Device         |One Plus 5|
|Android Version| Android Pie 9.0.0|

**Additional context**

@nikit19 @iamareebjamal Should I go for it?

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by angmas1**
Fixes #865 

Changes: Added check for invalid email address in attendee details.

Screenshots for the change:
whatsapp image 2019-02-09 at 16 39 20

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 1]
```diff
@@ -383,14 +386,28 @@ class AttendeeFragment : Fragment() {
             } else {
                 attendees.addAll(attendeeRecyclerAdapter.attendeeList)
             }
-            val country = if (country.text.isEmpty()) country.text.toString() else null
-            attendeeViewModel.createAttendees(attendees, country, paymentOptions[selectedPaymentOption])
 
-            attendeeViewModel.isAttendeeCreated.observe(this, Observer { isAttendeeCreated ->
-                if (isAttendeeCreated && selectedPaymentOption == paymentOptions.indexOf(getString(R.string.stripe))) {
-                    sendToken()
+            for (i in 0 until attendees.size) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Why do you need i?

Also, all logic should be in ViewModel

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Okay will close this request and post another one with logic in ViewModel

**Comment by iamareebjamal** [comment]
Please don't close PRs. I will not be reviewing any PRs which have been created more than once for an issue.

**Comment by iamareebjamal** [comment]
Push new code on this branch, there is no reason to close a PR

**Comment by angmas1** [comment]
Okay sorry.

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt [hunk 2]
```diff
@@ -395,6 +396,14 @@ class AttendeeViewModel(
             })
     }
 
+    fun checkEmail(attendees: ArrayList<Attendee>): Boolean {
+        (0 until attendees.size).forEach {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Why do you need the index of the list? This is not C

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Fixed

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt [hunk 3]
```diff
@@ -395,6 +396,15 @@ class AttendeeViewModel(
             })
     }
 
+    fun checkEmail(attendees: ArrayList<Attendee>): Boolean {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Change to areAttendeeEmailsValid

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: changed

**Pull request merged**